launch/launch/__init__.py

```diff
@@ -75,8 +78,11 @@ def __init__(self, cmd, name, output_handler, exit_handler, env=None):
 
     def send_signal(self, signal):
         if self.transport:
+            self.task_state.signals_received.append(signal)
             self.transport.send_signal(signal)
 
     def terminate(self):
-        if self.transport:
+        if os.name != 'nt':
+            self.send_signal(signal.SIGINT)
```

Comment:
I think that line 88 covers the case where the process doesn't exit when it receives SIGINT.
https://github.com/ros2/launch/pull/36/files#diff-95fa932db2912f5a4060524ea72cd0cfR88

The task descriptor's future will be considered finished, so the thread waiting in the `yield` statement in the `while True` loop in `launcher.py` will wake up and initialize exit.

https://github.com/ros2/launch/blob/master/launch/launch/launcher.py#L149

I was testing this manually but I will add a unit test for this case.